packages/host/app/components/ai-assistant/toast.gts

```diff
@@ -176,28 +177,24 @@ export default class AiAssistantToast extends Component<Signature> {
             lastMessage[1].eventId,
           ),
       )[0] ?? null;
-    if (lastMessage) {
+    let fifteenMinutesAgo = subMinutes(new Date(), 15);
+    if (lastMessage && isAfter(lastMessage[1].created, fifteenMinutesAgo)) {
       state.value = {
         roomId: lastMessage[0],
         message: lastMessage[1],
       };
 
-      resetStateValueId = resetStateValue();
-      on.cleanup(() => {
-        if (resetStateValueId) {
-          clearInterval(resetStateValueId);
-        }
-      });
+      pollTask(this, resetStateValue);
```

Comment:
I investigated the flaky test that @lukemelia  mentioned and I found that it occurred because the toast could disappear before we could click the button to open the session, causing the assertions to fail. The only way to fix it was by simulating the user activity of hovering over the toast to prevent it from disappearing. https://github.com/cardstack/boxel/pull/1397/files#diff-2be5a5cbf1f1ef1a820efc9e2fb3c1ed6020a7c09f3c28a68c7ee861c9d1fdcbR2135

However, this caused a `runTask` recursive issue when trying to reset the state value, which blocked the test because Ember tests wait for all scheduled tasks to finish before continuing. I found a solution using `pollTask`.